### ISSUE TITLE
Parsing error 

### ISSUE THREAD

**Issue by David100mark** [koalaman/shellcheck#1606]
Hey, 

i got an unexpected syntax error, i guess it is a bug. Could you check on this?
The error:
- SC1073
- My shellcheck version online: 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
![grafik](https://user-images.githubusercontent.com/37455012/59157961-c3c8b300-8ab3-11e9-8b31-234d8e7c9b95.png)

my Script:

#!/bin/bash
function ProgressBar () {
     let _progress=(${1} * 100 / ${2} * 100)/100
     let _done=(${_progress} * 4)/10
     let _left=40-$_done
     _fill=$(printf "%${_done}s")
     _empty=$(printf "%${_left}s")
     printf "\rProgress : [${_fill// /#}${_empty// /-}] ${_progress}%%"
}
```

#### Here's what shellcheck currently says:
$ shellcheck myscript
 
Line 1:
function ProgressBar () {
^-- SC1009: The mentioned syntax error was in this function.
                        ^-- SC1073: Couldn't parse this brace group. Fix to allow more checks.
 
Line 2:
     let _progress=(${1} * 100 / ${2} * 100)/100
                   ^-- SC1036: '(' is invalid here. Did you forget to escape it?
                   ^-- SC1056: Expected a '}'. If you have one, try a ; or \n in front of it.
                   ^-- SC1072: Missing '}'. Fix any mentioned problems and try again.

$


i would not expect any error. For me the syntax seems to be right. I also let it run and it works fine.

Thank you very much in advance
David

**Comment by Nightfirecat** [comment]
This sounds accurate. Bash itself says that line includes a syntax error. See the following:

```
$ cat << EOF > myscript.sh
#!/bin/bash
function ProgressBar () {
let _progress=(\${1} * 100 / \${2} * 100)/100
let _done=(\${_progress} * 4)/10
let _left=40-\$_done
_fill=\$(printf "%\${_done}s")
_empty=\$(printf "%\${_left}s")
printf "\rProgress : [\${_fill// /#}\${_empty// /-}] \${_progress}%%"
}
ProgressBar
EOF

$ ./myscript.sh
./myscript.sh: line 3: let: _progress=( * 100 /  * 100)/100: syntax error: operand expected (error token is "* 100 /  * 100)/100")
./myscript.sh: line 4: let: _done=( * 4)/10: syntax error: operand expected (error token is "* 4)/10")
./myscript.sh: line 5: let: _left=40-: syntax error: operand expected (error token is "-")
Progress : [] %
```

**Comment by koalaman** [comment]
This is indeed a bug. Here's an easier way to reproduce the issue:

```
let x=(1)
```

Bash accepts this just fine, while shellcheck says:
```
In foo line 1:
let x=(1)
^----^ SC2148: Tips depend on target shell and yours is unknown. Add a shebang.
^----^ SC2219: Instead of 'let expr', prefer (( expr )) .
    ^-- SC2034: x appears unused. Verify use (or export if used externally).
      ^-- SC1036: '(' is invalid here. Did you forget to escape it?$ shellcheck foo

      ^-- SC1088: Parsing stopped
```

Bash appears to use its assignment parsing mode for `let`, just like it does for `eval`, `export` and `declare`, thereby allowing parentheses 

Additionally -- and surprisingly -- it allows a suffix after what would have been an array assignment, e.g. `let x=(1)/1`. This has the weird side effect that bash accepts `let x=(1)/1` but not `let x=1/(1)`.
